staging/src/k8s.io/apiserver/pkg/server/storage/resource_encoding_config_compatibility.go

```diff
@@ -0,0 +1,109 @@
+package storage
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/component-base/version"
+)
+
+type ResourceSpec struct {
+	// Version indicates the Kubernetes version from which this GVR is decodable
+	K8sVersion semver.Version
+
+	// Version indicates the resource version for which support was added
+	Version string
```

Comment:
version is somewhat overloaded, is this like "v1beta1", "v1", etc?

Reply:
yeah. I thought to use `APIVersion` for this field but to me that usually includes the group too